clang/test/CodeGenSYCL/device_global_int_footer_header.cpp

```diff
@@ -55,6 +55,9 @@ int main() {
 device_global<int> Basic;
 // CHECK-FOOTER-NEXT: device_global_map::add((void *)&::Basic, "_Z5Basic");
 
+extern device_global<int> ExternDevGlobal;
+// CHECK-FOOTER-NOT: ExternDevGlobal
+
```

Comment:
Does it make sense to add cases for the other two conditions here?   CHECK-NOTs for locals, params, etc

Reply:
I think we have diagnostics (with tests) for invalid cases. This case is unique because it isn't an invalid case. We just don't want to add the extern declaration without definition